src/main/java/wolox/training/models/Book.java

```diff
@@ -108,6 +114,7 @@ public String getYear() {
     }
 
     public void setYear(String year) {
+        Preconditions.checkNotNull(year, "Illegal Argument, year is Null.");
```

Comment:
Another good precondition is to check the year is valid, like it's not from the future or a negative year.

Reply:
Fixed at https://github.com/wolox-training/yh-java/pull/11/commits/6d7dd31f8f886894e790e10b107b0968347e49a7 commit